### PR TITLE
#1721 Agreement doesn't work anymore

### PR DESCRIPTION
**Pull request by UWinch**
Agreement can no longer be calculated (for finished, as well as with unfinished docs).

**How to test manually**
* see Issue
* uncheck `limit to finished documents`
* check that agreement is re-calculated

**Automatic testing**
* [ ] PR includes unit tests

**Documentation**
* [ ] PR updates documentation

### CODE REVIEW

File: webanno-api-dao/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/dao/AnnotationSchemaServiceImpl.java [hunk 1]
```diff
@@ -961,9 +961,9 @@ public boolean upgradeCasIfRequired(Iterable<CAS> aCasIter, Project aProject)
         // Check if the current CAS already contains the required type system
         boolean upgradePerformed = false;
         for (CAS cas : aCasIter) {
-            CasStorageSession.get().assertWritingPermitted(cas);
             
             if (cas != null && isUpgradeRequired(cas, ts)) {
+                CasStorageSession.get().assertWritingPermitted(cas);
```
Comment (reckart) [inline webanno-api-dao/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/dao/AnnotationSchemaServiceImpl.java:966]: I don't think we should check if writing is permitted only if we actually detect that an update is necessary. Even if no upgrade is necessary that the system is in principle allowed to perform the write operation. Was moving this assert into the condition necessary to fix the issue?

Comment (UWinch) [reply webanno-api-dao/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/dao/AnnotationSchemaServiceImpl.java:966]: Yes, the cas was null at this point.

Comment (reckart) [reply webanno-api-dao/src/main/java/de/tudarmstadt/ukp/clarin/webanno/api/dao/AnnotationSchemaServiceImpl.java:966]: I'll separate the null check from the upgrade check. Null CASes should just be skipped.

**Comment by UWinch** [comment]
Jenkins, can you test this please?

**Pull request merged**